test/execution/atomics_test.cpp

```diff
@@ -0,0 +1,513 @@
+#include <random>
+#include <string>
+
+#include "common/macros.h"
+#include "common/worker_pool.h"
+#include "execution/ast/ast_node_factory.h"
+#include "execution/ast/context.h"
+#include "execution/compiler/codegen.h"
+#include "execution/compiler/compiler.h"
+#include "execution/compiler/function_builder.h"
+#include "execution/sema/error_reporter.h"
+#include "execution/tpl_test.h"
+#include "execution/util/region.h"
+#include "execution/util/region_containers.h"
+#include "execution/vm/module.h"
+#include "execution/vm/vm_defs.h"
+#include "test_util/multithread_test_util.h"
+
+namespace noisepage::execution::test {
+
+class AtomicsTest : public TplTest {
+ public:
+  AtomicsTest() : region_("atomics_test"), pos_() {}
+
+  util::Region *Region() { return &region_; }
+
+  const SourcePosition &EmptyPos() const { return pos_; }
+
+  util::Region region_;
+  SourcePosition pos_;
+};
+
+// NOLINTNEXTLINE
+TEST_F(AtomicsTest, AtomicAndOr1) {
+  // Setup the compilation environment
+  sema::ErrorReporter error_reporter(&region_);
+  ast::AstNodeFactory factory(&region_);
+  ast::Context context(&region_, &error_reporter);
+
+  auto src = std::string(
+      "fun atomic_and(dest: *uint8, mask: uint8) -> uint8 {\n"
+      "  var x = @atomicAnd(dest, mask)\n"
+      "  return x\n"
+      "}\n"
+      "fun atomic_or(dest: *uint8, mask: uint8) -> uint8 {\n"
+      "  var x = @atomicOr(dest, mask)\n"
+      "  return x\n"
+      "}");
+
+  // Compile it...
+  auto input = compiler::Compiler::Input("Atomic Definitions", &context, &src);
+  auto module = compiler::Compiler::RunCompilationSimple(input);
+  EXPECT_STREQ(error_reporter.SerializeErrors().c_str(), "");
```

Comment:
Use [`ErrorReporter::HasErrors()`](https://github.com/cmu-db/noisepage/blob/master/src/include/execution/sema/error_reporter.h#L66)

Reply:
I can remove these now.  This was a hack to get the error printed out in the log when there were errors in the test.  The other TPL-based tests just have the assert on not having a nullptr.